integration-tests/bats/sql-server-mcp.bats

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+load $BATS_TEST_DIRNAME/helper/query-server-common.bash
+
+make_repo() {
+  mkdir "$1"
+  cd "$1"
+  dolt init
+  cd ..
+}
+
+setup() {
+  skiponwindows "tests are flaky on Windows"
+  setup_no_dolt_init
+  make_repo repo1
+}
+
+teardown() {
+  stop_sql_server 1 && sleep 0.5
+  rm -rf $BATS_TMPDIR/sql-server-mcp-test$$
+  teardown_common
+}
+
+# Helper: wait for an MCP HTTP server to accept TCP connections on the given port
+# wait_for_mcp_port <PORT> <TIMEOUT_MS>
+wait_for_mcp_port() {
+  port=$1
+  timeout_ms=$2
+  end_time=$((SECONDS+($timeout_ms/1000)))
+  while [ $SECONDS -lt $end_time ]; do
+    nc -z localhost "$port" >/dev/null 2>&1 && return 0
+    sleep 1
+  done
+  return 1
+}
+
+@test "sql-server mcp: --mcp-database connects to specified database" {
+  cd repo1
+
+  # Create target database before starting server
+  dolt sql -q "CREATE DATABASE mcpdb"
+
+  MCP_PORT=$( definePORT )
+  start_sql_server_with_args --host 0.0.0.0 --mcp-port "$MCP_PORT" --mcp-user root --mcp-database mcpdb
+  run wait_for_mcp_port "$MCP_PORT" 8500
+  [ $status -eq 0 ]
+
+  # initialize and list via MCP
+  INIT_FILE=$BATS_TMPDIR/mcp_init_md_$$.json
+  OUT_INIT=$BATS_TMPDIR/mcp_out_init_md_$$.json
+  echo '{"jsonrpc":"2.0","id":"1","method":"initialize","params":{"clientInfo":{"name":"bats","version":"0.0.0"},"capabilities":{}}}' > "$INIT_FILE"
+  run bash -c "curl -sS -D $BATS_TMPDIR/mcp_headers_md_$$.txt -H 'Content-Type: application/json' --data-binary @'$INIT_FILE' http://127.0.0.1:${MCP_PORT}/ > '$OUT_INIT'"
+  [ $status -eq 0 ]
+  SESSION=$(grep -i '^Mcp-Session-Id:' $BATS_TMPDIR/mcp_headers_md_$$.txt | awk -F': ' '{print $2}' | tr -d '\r')
+  [ -n "$SESSION" ]
+
+  CALL_FILE=$BATS_TMPDIR/mcp_call_md_$$.json
+  OUT_CALL=$BATS_TMPDIR/mcp_out_call_md_$$.json
+  echo '{"jsonrpc":"2.0","id":"2","method":"tools/call","params":{"name":"list_databases","arguments":{}}}' > "$CALL_FILE"
+  run bash -c "curl -sS -H 'Content-Type: application/json' -H 'Mcp-Session-Id: '$SESSION --data-binary @'$CALL_FILE' http://127.0.0.1:${MCP_PORT}/ > '$OUT_CALL'"
+  [ $status -eq 0 ]
+  run grep -E "mcpdb" "$OUT_CALL"
+  [ $status -eq 0 ]
+}
+
+@test "sql-server mcp: --mcp-user authenticates as specified sql user" {
+    cd repo1
+
+    # TODO: this is currently broken in bats for some reason,
+    # need to figure out why
+    # Create a sql user with no password and read privileges
+    # dolt sql -q "CREATE USER mcpuser@'%'"
+    # dolt sql -q "GRANT ALL PRIVILEGES ON *.* TO mcpuser@'%'"
```

Comment:
This fails in bats for some reason where when trying to connect as the created  user, the db returns the error: No auth methods defined for user. This does not happen outside of the bats context though.